src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -121,46 +128,127 @@ public bool GetWordVector(ref DvText word, float[] wordVector)
+        /// </summary>
+        public sealed class ColumnInfo
+        {
+            public readonly string Input;
+            public readonly string Output;
+
+            public ColumnInfo(string input, string output)
+            {
+                Contracts.CheckNonEmpty(input, nameof(input));
+                Contracts.CheckNonEmpty(output, nameof(output));
+
+                Input = input;
+                Output = output;
+            }
+        }
+
         private const string RegistrationName = "WordEmbeddings";
 
         private const int Timeout = 10 * 60 * 1000;
 
         /// <summary>
-        /// Public constructor corresponding to <see cref="SignatureDataTransform"/>.
+        /// Instantiates <see cref="WordEmbeddingsTransform"/> using the pretrained word embedding model specified by <paramref name="modelKind"/>.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="inputColumn">Name of the input column.</param>
+        /// <param name="outputColumn">Name of the output column.</param>
+        /// <param name="modelKind">The pretrained word embedding model.</param>
+        public WordEmbeddingsTransform(IHostEnvironment env, string inputColumn, string outputColumn,
+           PretrainedModelKind modelKind = PretrainedModelKind.Sswe)
+           : this(env, modelKind, new ColumnInfo(inputColumn, outputColumn))
+        {
+        }
+
+        /// <summary>
+        /// Instantiates <see cref="WordEmbeddingsTransform"/> using the custom word embedding model by loading it from the file specified by the <paramref name="customLookupTable"/>.
         /// </summary>
-        public WordEmbeddingsTransform(IHostEnvironment env, Arguments args, IDataView input)
-                : base(env, RegistrationName, Contracts.CheckRef(args, nameof(args)).Column,
-                     input, TestIsTextVector)
+        /// <param name="env">Host Environment.</param>
+        /// <param name="inputColumn">Name of the input column.</param>
+        /// <param name="outputColumn">Name of the output column.</param>
+        /// <param name="customLookupTable">Filename for custom word embedding model.</param>
+        public WordEmbeddingsTransform(IHostEnvironment env, string inputColumn, string outputColumn, string customLookupTable)
+           : this(env, customLookupTable, new ColumnInfo(inputColumn, outputColumn))
         {
+        }
+
+        /// <summary>
+        /// Instantiates <see cref="WordEmbeddingsTransform"/> using the pretrained word embedding model specified by <paramref name="modelKind"/>.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="modelKind">The pretrained word embedding model.</param>
+        /// <param name="columns">Input/Output columns.</param>
+        public WordEmbeddingsTransform(IHostEnvironment env, PretrainedModelKind? modelKind, params ColumnInfo[] columns)
```

Comment:
>PretrainedModelKind? modelKind [](start = 61, length = 30)

can you push modelKind and customLookup table to ColumnInfo class?

Reply:
Not I don't think so because modelKind and customLookupTable are same for all the columns and only needed once.

---
In reply to: [218250633](https://github.com/dotnet/machinelearning/pull/928#discussion_r218250633) [](ancestors = 218250633)

Reply:
Ok, at least don't make it nullable. 


---
In reply to: [218251539](https://github.com/dotnet/machinelearning/pull/928#discussion_r218251539) [](ancestors = 218251539,218250633)